configuration/src/test/java/io/airlift/configuration/TestConfigurationLoader.java

```diff
@@ -60,46 +61,55 @@ public void testLoadsFromSystemProperties()
 
         assertEquals(properties.get("test"), "foo");
 
-        System.getProperties().remove("test");
+        System.clearProperty("test");
     }
 
     @Test
     public void testLoadsFromFile()
             throws IOException
     {
-        final File file = File.createTempFile("config", ".properties", tempDir);
-        PrintStream out = new PrintStream(new FileOutputStream(file));
-        try {
-            out.print("test: foo");
-        }
-        catch (Exception e) {
-            out.close();
-        }
-
+        File file = createConfigFile(out -> out.print("test: foo"));
         System.setProperty("config", file.getAbsolutePath());
 
         Map<String, String> properties = loadProperties();
 
         assertEquals(properties.get("test"), "foo");
         assertEquals(properties.get("config"), file.getAbsolutePath());
 
-        System.getProperties().remove("config");
+        System.clearProperty("config");
+    }
+
+    @Test
+    public void testTrimWhitespaceFromFile()
+            throws IOException
+    {
+        File file = createConfigFile(out -> {
+            out.println(" \t trim-whitespace-key1 \t =  \t key1-value \t ");
+            out.println(" \t trim-whitespace-key2 \t =  \t key2-value \t ");
+        });
+        System.setProperty("config", file.getAbsolutePath());
+        System.setProperty("trim-whitespace-key2", " \t key2-value \t ");
+
+        Map<String, String> properties = loadProperties();
+
+        // config files are often human-managed and we need to be permissive, stripping trailing whitespace if any
+        assertEquals(properties.get("trim-whitespace-key1"), "key1-value");
+
+        // trailing whitespace in JVM property value is not so likely to be unintentional, so preserve it
+        assertEquals(properties.get("trim-whitespace-key2"), " \t key2-value \t ");
```

Comment:
I put it here so that it's "documented".
I'm not sure I want this behavior. Maybe we should trim always.

@martint @willmo83